### PR TITLE
[TRTLLM-12385][feat] Add VisualGen LPIPS evaluation script

### PR DESCRIPTION
**Pull request by yibinl-nvidia**
## Summary by CodeRabbit

* **Chores**
  * Added development dependencies and evaluation infrastructure for video quality assessment using LPIPS metrics to compare generated outputs against reference videos, enabling comprehensive visual quality validation throughout development and testing workflows.

[Review Change Stack](https://app.coderabbit.ai/change-stack/NVIDIA/TensorRT-LLM/pull/14325?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: scripts/visualgen_eval/lpips_video_utils.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+import torch
+
+
+def _video_tensor_to_lpips_batch(
+    video: torch.Tensor,
+    device: str,
+) -> torch.Tensor:
+    video = video.detach().cpu()
+    if video.dim() == 5:
+        video = video[0]
```
[the remark this answers is not in the document]
Comment (yibinl-nvidia) [reply scripts/visualgen_eval/lpips_video_utils.py]: not applicable.

File: scripts/visualgen_eval/lpips_video_utils.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    batch = frames.clamp(0, 1).permute(0, 3, 1, 2)
+    return batch.to(device=device, dtype=torch.float32) * 2.0 - 1.0
+
+
+def _decode_video_to_lpips_batch(
+    video_path: Path,
+    device: str,
+) -> torch.Tensor:
+    if not video_path.exists():
+        raise FileNotFoundError(f"Golden video not found: {video_path}")
```
[the remark this answers is not in the document]
Comment (yibinl-nvidia) [reply scripts/visualgen_eval/lpips_video_utils.py]: file updated.

File: scripts/visualgen_eval/visual_gen_lpips_score_eval.py [hunk 3]
```diff
@@ -0,0 +1,765 @@
...
+        sample_params = defaults | _extract_generation_params(sample)
+        samples.append(
+            {
+                "id": sample_id,
+                "prompt": prompt,
+                "media_type": media_type,
+                "reference_path": reference_path,
+                "generated_path": generated_path,
+                "output_media": output_media,
+                "params": sample_params,
```
[the remark this answers is not in the document]
Comment (yibinl-nvidia) [reply scripts/visualgen_eval/visual_gen_lpips_score_eval.py]: fixed

File: scripts/visualgen_eval/visual_gen_lpips_score_eval.py [hunk 4]
```diff
@@ -0,0 +1,765 @@
...
+            visual_gen.shutdown()
+        temp_video_dir.cleanup()
+        gc.collect()
+        if torch.cuda.is_available():
+            torch.cuda.empty_cache()
+
+    scores = [sample["lpips_score"] for sample in sample_results]
+    mean_score = float(sum(scores) / len(scores))
+    passed = None if threshold is None else mean_score < threshold
+    return {
```
[the remark this answers is not in the document]
Comment (yibinl-nvidia) [reply scripts/visualgen_eval/visual_gen_lpips_score_eval.py]: fixed

File: scripts/visualgen_eval/visual_gen_lpips_score_eval.py [hunk 5]
```diff
@@ -0,0 +1,765 @@
...
+    if output_media is None:
+        output_path = pathlib.Path(f"{sample_id}{default_suffix}")
+    else:
+        output_path = pathlib.Path(output_media).expanduser()
+
+    if output_path.is_absolute():
+        return output_path
+    if output_dir is not None:
+        return output_dir / output_path
+    return dataset_dir / output_path
```
[the remark this answers is not in the document]
Comment (yibinl-nvidia) [reply scripts/visualgen_eval/visual_gen_lpips_score_eval.py]: fixed

**Comment by yibinl-nvidia** [comment]
close this as https://github.com/NVIDIA/TensorRT-LLM/pull/13567 merged.
